Source/WebCore/testing/MockGamepadProvider.cpp

```diff
@@ -106,8 +120,14 @@ bool MockGamepadProvider::disconnectMockGamepad(unsigned index)
 
     m_connectedGamepadVector[index] = nullptr;
 
-    for (auto& client : m_clients)
-        client->platformGamepadDisconnected(*m_mockGamepadVector[index]);
+    auto gamepadToRemove = m_mockGamepadVector[index].get();
+    for (auto& client : m_clients) {
+        if (m_invisibleGamepadsForClient.contains(client) && m_invisibleGamepadsForClient.get(client).contains(gamepadToRemove)) {
+            auto itr = m_invisibleGamepadsForClient.find(client);
+            itr->value.remove(gamepadToRemove);
```

Comment:
I know this is really nit-picky, but please use `auto iter` instead of `auto itr` here as the former is much more commonly used in WebKit project sources.

You could also combine the two lines into a single line:
```
             m_invisibleGamepadsForClient.find(client)->value.remove(gamepadToRemove);
```